src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -111,7 +111,7 @@ public async Task SynchronizeWithServer()
                 }
                 else
                 {
-                    var analysisInformation = await GetAnalysisInformation(projectKey);
+                    var analysisInformation = await GetAnalysisInformation(projectKey, serverBranch);
```

Comment:
server branch is what we get from IStatefulServerBranchProvider.GetServerBranchNameAsync -- this will return Main if it can't find branch information, right?

Reply:
Unfortunately, no. It will return `null` in that case, which the SonarQube APIs will then interpret as meaning "fetch whatever branch is marked as the main branch".

I think we should update `IServerBranchProvider`: move the logic from the taint synchronizer that locates the main branch to `IServerBranchProvider` so it can explicitly return the name of the branch to fetch in all cases.

Reply:
Ok then I will add a fallback to main branch 👍